File: codeup/100000575/G.py
#!/usr/bin/python3
import sys

while True:
    line = sys.stdin.readline()
    if not line:
        break
    A = [0 for i in range(6)]
    num_list = [int(x) for x in line.split()]
    my_flag = 1
    A4_cnt = 0
    A2_flag = 0
    for num in num_list[1:]:
        temp = num % 5
        if temp == 0:
            if num % 2 == 0:
                A[1] += num
        elif temp == 1:
            A[2] += num*my_flag
            my_flag *= -1
            A2_flag = 1
        elif temp == 2:
            A[3] += 1
        elif temp == 3:
            A[4] += num
            A4_cnt += 1
        elif temp == 4:
            if A[5] < num:
                A[5] = num

    for i in range(1, 6):
        if A[i] and i == 4:
            print("{:.1f}".format(1.0 * A[4] / A4_cnt), end=' ')
        elif i == 5:
            if A[i] == 0:
                print("N")
            else:
                print(A[i])
        elif i == 2:
            if A2_flag == 0:
                print("N", end=' ')
            else:
                print(A[i], end=' ')
        else:
            if A[i] == 0:
                print("N", end=' ')
            else:
                print(A[i], end=' ')

File: codeup/100000575/G.cpp
#include <iostream>
using namespace std;

int main() {
  int A[6], N, num, flag, A2_flag, A4_cnt;
  while (scanf("%d", &N) != EOF) {
    fill(A + 1, A + 6, 0);
    flag = 1, A2_flag = A4_cnt = 0;
    while (N--) {
      scanf("%d", &num);
      switch (num % 5) {
        case 0:
          if (num % 2 == 0) A[1] += num;
          break;
        case 1:
          A[2] += num * flag;
          flag *= -1;
          A2_flag = 1;
          break;
        case 2:
          A[3]++;
          break;
        case 3:
          A[4] += num;
          A4_cnt++;
          break;
        case 4:
          if (A[5] < num) A[5] = num;
          break;
        default:
          break;
      }
    }
    for (int i = 1; i < 6; i++) {
      switch (i) {
        case 1:
        case 3:
          if (A[i] == 0)
            printf("N ");
          else
            printf("%d ", A[i]);
          break;
        case 2:
          if (A2_flag == 0)
            printf("N ");
          else
            printf("%d ", A[i]);
          break;
        case 4:
          if (A[i] == 0)
            printf("N ");
          else
            printf("%.1lf ", A[i] * 1.0 / A4_cnt);
          break;
        case 5:
          if (A[i] == 0)
            printf("N\n");
          else
            printf("%d\n", A[i]);
          break;
        default:
          break;
      }
    }
  }

  return 0;
}